juju/juju
# Introduce auto bootstrap to kubeconfig contexts.

This change introduces changes to Juju that will allow a user to
bootstrap to a Kubernetes cluster from the kubeconfig file without the
add-k8s step.

## Checklist


## QA steps

This change allows you to bootstrap directly to Kubernetes context's defined in your local KubeConfig file.

This can be tested by making a Kubeconfig that points to several clusters. 
 - GKE
 - Minikube
 - Kind
 - Azure.

Then run `juju clouds` to make sure the clouds appear.

After this `juju bootstrap <context_name>` can be run.

## Documentation changes

Discourse post incoming

## Bug reference

NA
